include/rocksdb/options.h

```diff
@@ -2172,6 +2172,23 @@ struct IngestExternalFileOptions {
   //
   // XXX: "bottommost" is obsolete/confusing terminology to refer to last level
   bool fail_if_not_bottommost_level = false;
+
+  // Indicate that files being ingested are all from some live DB.
+  // If set to true, ingestion will
+  // - not delete original files,
+  // - allow the files to not be generated by SstFileWriter, and
+  // - ignore cf_id mismatch between cf_id in the files and the CF they are
+  // being ingested into.
+  // Warning: This ONLY works for SST files where all keys have sequence number
+  // zero and with no duplicated user keys (this should be guaranteed if the
```

Comment:
This `MoveColumnFamily` API is used to merge (ingest) one column family's data into another column family in the same DB, right? I wonder how snapshots held by other column families could affect the data preprocessing of the ingestion preparation column family. Since it needs to do a full compaction to get rid of duplicate user keys, does that mean all snapshots should be released before this full compaction. And maybe that's requirement is not straightforward to guarantee when there are other column families.

Reply:
Yeah, snapshot can prevent such keys to be dropped. Maybe the temporary CF should in a temporary DB..